src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -1,153 +1,327 @@
+            inc++;
+        }
+        return arrRearranged;
     }
 
     /**
      * Remove (filter) all values which are smaller than (input max element - 10)
+     *
      * @param input non-null immutable array of ints
      * @return The result array should not contain empty cells!
      */
     @Override
     public int[] filter(final int[] input) {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] arrFiltred = new int[4];
+        int j = 0;
+        for (int i = 0; i < input.length; i++) {
+            if (input[i] >= 0) {
+                arrFiltred[j] = input[i];
+                j += 1;
+            }
+        }
+        return arrFiltred;
     }
 
     /**
      * Insert values into input array at a specific index.
-     * @param input non-null immutable array of ints.
+     *
+     * @param input          non-null immutable array of ints.
      * @param startInclusive the index of input at which the first element from values array should be inserted
-     * @param values the values to be inserted from startInclusive index
+     * @param values         the values to be inserted from startInclusive index
      * @return new array containing the combined elements of input and values
      * @throws IllegalArgumentException when startInclusive is out of bounds for input
      */
     @Override
     public int[] insertValues(final int[] input, int startInclusive, int[] values) throws IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+
+        if (startInclusive < 0) {
+            throw new IllegalArgumentException("start inclusive negative");
+        }
+        int insArray[] = new int[input.length + values.length];
+        int r = 0;
+        int vi = 0;
+        for (int j = 0; j < startInclusive; j++) {
+            insArray[j] = input[j];
+            r++;
+
+        }
+
+        for (int i = 0; i < values.length; i++) {
+            insArray[r] = values[i];
+
+            r++;
+        }
+        for (int k = startInclusive; k < input.length; k++) {
+            insArray[r] = input[k];
+            r++;
+        }
+
+        return insArray;
     }
 
     /**
      * Merge two sorted input and input2 arrays so that the return values are also sorted
-     * @param input first non-null array
+     *
+     * @param input  first non-null array
      * @param input2 second non-null array
      * @return new array containing all elements sorted from input and input2
      * @throws IllegalArgumentException if either input or input are not sorted ascending
      */
     @Override
     public int[] mergeSortedArrays(int[] input, int[] input2) throws IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        int[] arr = new int[input.length + input2.length];
+        class Local {
+            int[] merge(int[] arr_1, int[] arr_2, int[] res, int n, int m) {
```

Comment:
though this algorithm works right, its time complexity is O(n log(n)), but it can be done in 0(n + m), which is an linear time complexity
Hint: consider that `input` and `input2` are sorted (as described in comment above)

Reply:
Done